src/record.rs

```diff
@@ -99,13 +122,96 @@ impl<'a> Serializable<'a> for MultiPlexedRecord<'a> {
                 position,
                 payload,
             }),
+            RecordType::AppendRecords => Some(MultiPlexedRecord::AppendRecords {
+                queue,
+                position,
+                records: MultiRecord::new(payload).ok()?,
+            }),
             RecordType::Truncate => Some(MultiPlexedRecord::Truncate { queue, position }),
             RecordType::Touch => Some(MultiPlexedRecord::Touch { queue, position }),
             RecordType::DeleteQueue => Some(MultiPlexedRecord::DeleteQueue { queue, position }),
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct MultiRecord<'a> {
+    buffer: &'a [u8],
+    byte_offset: usize,
+}
+
+impl<'a> MultiRecord<'a> {
+    pub fn new(buffer: &[u8]) -> Result<MultiRecord, MultiRecordCorruption> {
+        let mut mrecord = MultiRecord {
+            buffer,
+            byte_offset: 0,
+        };
+        // verify the content is not corrupted
+        for record in mrecord {
+            record?;
+        }
+
+        mrecord.reset_position();
+
+        Ok(mrecord)
+    }
+
+    pub fn new_unchecked(buffer: &[u8]) -> MultiRecord {
+        MultiRecord {
+            buffer,
+            byte_offset: 0,
+        }
+    }
+
+    pub fn serialize<'b, T: Iterator<Item = &'b [u8]>>(items: T, mut position: u64) -> Vec<u8> {
+        let mut res = Vec::new();
+        for item in items {
+            assert!(item.len() <= u16::MAX as usize);
+            res.extend_from_slice(&position.to_le_bytes());
+            position += 1;
+            res.extend_from_slice(&(item.len() as u16).to_le_bytes());
```

Comment:
can we serialize a position with all the records? (even if it is autoincrement for the moment)

The point is that I strongly suspect for different reason we will have to let the user more freedom on its declaration of position (as long as they are monotonically increasing).

Adding position here does not hurt, and will help backward compat.

Reply:
there is already an explicit position with each record:
```rust
res.extend_from_slice(&position.to_le_bytes());
position += 1;
```
I'll add a function to use arbitrary position instead of autoincrement